package com.enaboapps.switchify.screens.settings.shared

import androidx.compose.runtime.Composable
import com.enaboapps.switchify.R
import com.enaboapps.switchify.components.PreferenceValueSelector
import com.enaboapps.switchify.service.techniques.pointscan.PointScanSettings
import com.enaboapps.switchify.service.utils.ContinuousLineSpeedUtils

@Composable
fun SinglePointScanSpeedStepper() {
    PreferenceValueSelector(
        value = PointScanSettings.getLineSpeedLevel(),
        titleResId = R.string.preference_title_single_point_scan_speed,
        summaryResId = R.string.preference_summary_single_point_scan_speed,
        values = ContinuousLineSpeedUtils.getPresetOptions()
            .map { it.representativeLevel }
            .toIntArray(),
        buttonLabelFormatter = { speedLevel -> PointScanSettings.getSpeedLevelDescription(speedLevel) },
        displayFormatter = { speedLevel -> PointScanSettings.getSpeedLevelDescription(speedLevel) },
        onValueChanged = { newSpeedLevel ->
            PointScanSettings.setLineSpeedLevel(newSpeedLevel)
        }
    )
}
